samcli/commands/local/lib/sam_api_provider.py

```diff
@@ -525,3 +403,249 @@ def _normalize_binary_media_type(value):
             return None
 
         return value.replace("~1", "/")
+
+
+class AbstractParserProvider(object):
```

Comment:
Can you explain the difference of the ParserProvider vs the ApiProvider? This looks overkill to me without understanding intentions of the split.